prov/sockets/src/sock_poll.c

```diff
@@ -141,13 +141,19 @@ static int sock_poll_poll(struct fid_poll *pollset, void **context, int count)
 			cntr = container_of(list_item->fid, struct sock_cntr,
 						cntr_fid);
 			sock_cntr_progress(cntr);
-			pthread_mutex_lock(&cntr->mut);
-			if (atomic_get(&cntr->value) >=
-				atomic_get(&cntr->threshold)) {
-				*context++ = cntr->cntr_fid.fid.context;
-				ret_count++;
+			{
+				int val = atomic_get(&cntr->value);
+				int old_val = atomic_get(&cntr->old_value);
+				if (val > old_val) {
+					*context++ = cntr->cntr_fid.fid.context;
+					ret_count++;
+					int new_val = old_val;
+					do {
+						old_val = new_val;
+						new_val = atomic_cswap(&cntr->old_value, old_val, val);
+					} while(new_val != old_val && new_val < val);
+				}
```

Comment:
This part might not be needed as part of this patch. I think fi_poll() with counters need some clarification (please see https://github.com/ofiwg/libfabric/issues/2240). May be we can drop this (along with the atomic_cswap implementation) from the PR.